walker.go

```diff
@@ -51,15 +49,18 @@ const (
 	countHashes      = "file-hash-count"
 )
 
+// WriteCallback is a function to write a Walk.
+type WriteCallback func(*fspb.Walk) error
```

Comment:
Maybe call it `FinalizeWalkCallback`, `WalkCallback` or `RunWalkCallback` at this point?
One might use it to check the final walk without ever writing it anywhere.

If you take the suggestion, I'd add more to the comment to describe what can be done with it:

```go
// RunWalkCallback is called by Walker at the end of the Run.
// The callback is typically used to dump the walk to disk and/or perform any other checks.
type RunWalkCallback func(*fspb.Walk) error
```

Reply:
Thanks, that sounds good! I've incorporated that.